puppet.go

```diff
@@ -354,19 +354,24 @@ func (puppet *Puppet) updateAvatar(ctx context.Context, source *User, info *type
 			return true
 		}
 		avatarContentType = http.DetectContentType(avatarData)
+		puppet.AvatarPath = info.Profile.AvatarPath
+		hash := sha256.Sum256(avatarData)
+		newHash := hex.EncodeToString(hash[:])
+		if puppet.AvatarHash == newHash && puppet.AvatarSet {
+			log.Debug().
+				Str("avatar_hash", newHash).
+				Str("new_avatar_path", puppet.AvatarPath).
+				Msg("Avatar path changed, but hash didn't")
+			// Path changed, but actual avatar didn't
+			return true
+		}
+		puppet.AvatarHash = newHash
+		source.Client.Store.ContactStore.StoreContact(ctx, *info)
+		err = source.Client.Store.ContactStore.StoreContact(ctx, *info)
+		if err != nil {
```

Comment:
I don't really get what this whole PR is changing. It removes an unnecessary hash check for contact avatars, but that shouldn't break anything even if it's there. These StoreContact calls also shouldn't be here, but maybe they're somehow related to the actual fix?

Reply:
This was supposed to fix a discrepancy in how the `Puppet` & `Contact` structs stored the avatar hash, which could cause the puppet-update function to mistakenly think the avatar didn't change when it really did.

However, that seems to have been fixed ever since the `Contact` stopped storing the avatar hash.  This also makes my `StoreContact` call obsolete now, so you're correct in saying it shouldn't be there.

With that said, I'm still seeing avatar updates get missed due to aggressive caching of fetched profile info, but I'll open a different issue for that.